File: Tuples/tuples.c
#include <stdio.h>
#include <time.h>

typedef struct {
    int entero;
    float flotante;
    char cadena[50];
} Tupla;

int main() {
    // Creación de la "tupla"
    Tupla tup = {123, 456.789, "Hello, World!"};

    // Medición de tiempo de acceso
    clock_t start_time = clock();
    for(int i = 0; i < 1000000; i++) {
        volatile int entero = tup.entero;
        volatile float flotante = tup.flotante;
        volatile char cadena = tup.cadena[0];
    }
    
    double elapsed_time = (double)(clock() - start_time) / CLOCKS_PER_SEC * 1000;
    printf("Tiempo de acceso: %.2f ms\n", elapsed_time);

    return 0;
}

File: Tuples/tuples.py
import time

# Creación de la tupla
tup = (123, 456.789, "Hello, World!")

# Medición de tiempo de acceso
start_time = time.time()
for _ in range(1000000):
    _ = tup[0]
    _ = tup[1]
    _ = tup[2]

elapsed_time = (time.time() - start_time) * 1000  # Convertir a milisegundos
print(f"Tiempo de acceso: {elapsed_time:.2f} ms")
